common/src/main/proto/CorfuProtocol.proto

```diff
@@ -87,20 +84,23 @@ message PingRequest {
 message PingResponse {
 }
 
+// TODO(Zach): Update handshake logic
 message AuthenticateRequest {
-
+    UUID clientId = 1;
+    UUID serverId = 2;
 }
 
+// TODO(Zach): Update handshake logic
 message AuthenticateResponse {
-
+    UUID serverId = 1;
+    string corfuVersion = 2;
```

Comment:
You can remove this and do the version check on every message instead. The message version can be kept inside the header.

Reply:
I'll incorporate this in a separate PR later on.